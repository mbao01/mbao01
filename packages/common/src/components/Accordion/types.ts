import * as AccordionPrimitive from "@radix-ui/react-accordion";

export type AccordionProps = React.ComponentPropsWithoutRef<typeof AccordionPrimitive.Root>;

export type AccordionContentProps = React.ComponentPropsWithoutRef<
  typeof AccordionPrimitive.Content
>;

export type AccordionItemProps = React.ComponentPropsWithoutRef<typeof AccordionPrimitive.Item>;

export type AccordionTriggerProps = React.ComponentPropsWithoutRef<
  typeof AccordionPrimitive.Trigger
>;
