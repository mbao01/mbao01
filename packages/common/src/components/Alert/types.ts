import { type VariantProps } from "../../libs";
import { getAlertClasses } from "./constants";

export type AlertProps = React.HTMLAttributes<HTMLDivElement> &
  VariantProps<typeof getAlertClasses>;

export type AlertTitleProps = React.HTMLAttributes<HTMLHeadingElement>;

export type AlertDescriptionProps = React.HTMLAttributes<HTMLDivElement>;
