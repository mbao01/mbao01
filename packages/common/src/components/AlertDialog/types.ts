import * as AlertDialogPrimitive from "@radix-ui/react-alert-dialog";
import { type VariantProps } from "../../libs";
import { getButtonClasses } from "../Button/constants";
import {
  getDialogContentClasses,
  getDialogDescriptionClasses,
  getDialogFooterClasses,
  getDialogHeaderClasses,
  getDialogOverlayClasses,
  getDialogTitleClasses,
} from "../Dialog/constants";

export type AlertDialogProps = React.ComponentPropsWithoutRef<typeof AlertDialogPrimitive.Root>;

export type AlertDialogOverlayProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Overlay
> &
  VariantProps<typeof getDialogOverlayClasses>;

export type AlertDialogContentProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Content
> &
  VariantProps<typeof getDialogContentClasses>;

export type AlertDialogHeaderProps = React.HTMLAttributes<HTMLDivElement> &
  VariantProps<typeof getDialogHeaderClasses>;

export type AlertDialogFooterProps = React.HTMLAttributes<HTMLDivElement> &
  VariantProps<typeof getDialogFooterClasses>;

export type AlertDialogTitleProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Title
> &
  VariantProps<typeof getDialogTitleClasses>;

export type AlertDialogDescriptionProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Description
> &
  VariantProps<typeof getDialogDescriptionClasses>;

export type AlertDialogActionProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Cancel
> &
  VariantProps<typeof getButtonClasses>;

export type AlertDialogCancelProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Cancel
> &
  VariantProps<typeof getButtonClasses>;
