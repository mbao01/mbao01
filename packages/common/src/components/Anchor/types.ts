import { VariantProps } from "../../libs";
import { getAnchorClasses } from "./constant";

export type AnchorProps = React.AnchorHTMLAttributes<HTMLAnchorElement> & {
  isExternal?: boolean;
} & VariantProps<typeof getAnchorClasses>;
