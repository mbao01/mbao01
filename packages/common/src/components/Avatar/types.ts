import * as AvatarPrimitive from "@radix-ui/react-avatar";
import { type VariantProps } from "../../libs";
import {
  getAvatarClasses,
  getAvatarFallbackClasses,
  getAvatarImageClasses,
} from "./constants";

export type AvatarProps = Omit<
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Root>,
  "asChild" | "size"
> &
  VariantProps<typeof getAvatarClasses>;

export type AvatarImageProps = Omit<
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Image>,
  "asChild"
> &
  VariantProps<typeof getAvatarImageClasses>;

export type AvatarFallbackProps = Omit<
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Fallback>,
  "asChild"
> &
  VariantProps<typeof getAvatarFallbackClasses>;
