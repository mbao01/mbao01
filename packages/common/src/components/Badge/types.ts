import { type VariantProps } from "../../helpers";
import { getBadgeClasses } from "./constants";

export type BadgeProps = React.HTMLAttributes<HTMLSpanElement> &
  VariantProps<typeof getBadgeClasses>;
