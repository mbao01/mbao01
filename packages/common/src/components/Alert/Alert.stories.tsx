import type { Meta, StoryContext, StoryFn, StoryObj } from "@storybook/react";
import { InfoIcon } from "lucide-react";
import { Alert } from "./Alert";
import { type AlertProps } from "./types";

const withAlert = (_: StoryFn, context: StoryContext<AlertProps>) => {
  return (
    <Alert {...context.args}>
      <InfoIcon className="h-4 w-4" />
      <Alert.Title>Heads up!</Alert.Title>
      <Alert.Description>You can add components to your app using the cli.</Alert.Description>
    </Alert>
  );
};

// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
  component: Alert,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
    layout: "centered",
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
  tags: ["autodocs"],
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {},
  decorators: [withAlert],
} satisfies Meta<typeof Alert>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default: Story = {
  args: {},
};

export const PrimaryAlert: Story = {
  args: {
    variant: "primary",
  },
};

export const SuccessAlert: Story = {
  args: {
    variant: "success",
  },
};

export const OutlineAlert: Story = {
  args: {
    outline: true,
    variant: "error",
  },
};
