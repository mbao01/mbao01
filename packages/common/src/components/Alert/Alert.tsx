import type { AlertDescriptionProps, AlertProps, AlertTitleProps } from "./types";
import { cn } from "../../utilities";
import { getAlertClasses } from "./constants";

const Alert = ({ variant, outline, className, ...props }: AlertProps) => (
  <div role="alert" className={cn(getAlertClasses({ variant, outline }), className)} {...props} />
);

const AlertTitle = ({ className, ...props }: AlertTitleProps) => (
  <h5 className={cn("mb-1 font-medium leading-none tracking-tight", className)} {...props} />
);

const AlertDescription = ({ className, ...props }: AlertDescriptionProps) => (
  <div className={cn("text-sm [&_p]:leading-relaxed", className)} {...props} />
);

Alert.Title = AlertTitle;
Alert.Description = AlertDescription;

export { Alert };
