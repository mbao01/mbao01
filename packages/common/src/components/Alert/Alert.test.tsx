import { render, screen } from "@testing-library/react";
import { type AlertProps } from "./types";
import { Alert } from "./Alert";

describe("Alert", () => {
  const renderAlert = (heading: string, props?: AlertProps) =>
    render(
      <Alert {...props}>
        <Alert.Title>{heading}</Alert.Title>
        <Alert.Description>You should pay attention to me</Alert.Description>
      </Alert>
    );
  it("renders a basic alert", () => {
    renderAlert("Hey!");

    expect(screen.getByRole("heading", { name: "Hey!" })).toBeInTheDocument();
    expect(
      screen.getByText("You should pay attention to me")
    ).toBeInTheDocument();
  });

  it.each([
    ["has outline", true],
    ["has no outline", false],
  ] as const)("%s", (description, outline) => {
    const heading = `Alert ${description}`;
    const { asFragment } = renderAlert(heading, { outline });

    expect(screen.getByRole("heading", { name: heading })).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it.each([
    "neutral",
    "primary",
    "secondary",
    "accent",
    "ghost",
    "info",
    "success",
    "warning",
    "error",
  ] as const)("has %s variant", (variant) => {
    const heading = `Alert ${variant}`;
    const { asFragment } = renderAlert(heading, { variant });

    expect(screen.getByRole("heading", { name: heading })).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
