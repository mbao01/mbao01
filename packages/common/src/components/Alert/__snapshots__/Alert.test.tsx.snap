// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert > has accent variant 1`] = `
<DocumentFragment>
  <div
    class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg]:absolute [&>svg]:left-4 [&>svg]:text-inherit bg-accent text-accent-content border-accent"
    role="alert"
  >
    <h5
      class="mb-1 font-medium leading-none tracking-tight"
    >
      Alert accent
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      You should pay attention to me
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert > has error variant 1`] = `
<DocumentFragment>
  <div
    class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg]:absolute [&>svg]:left-4 [&>svg]:text-inherit bg-error text-error-content border-error"
    role="alert"
  >
    <h5
      class="mb-1 font-medium leading-none tracking-tight"
    >
      Alert error
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      You should pay attention to me
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert > has ghost variant 1`] = `
<DocumentFragment>
  <div
    class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg]:absolute [&>svg]:left-4 [&>svg]:text-inherit bg-ghost text-ghost-content border-transparent"
    role="alert"
  >
    <h5
      class="mb-1 font-medium leading-none tracking-tight"
    >
      Alert ghost
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      You should pay attention to me
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert > has info variant 1`] = `
<DocumentFragment>
  <div
    class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg]:absolute [&>svg]:left-4 [&>svg]:text-inherit bg-info text-info-content border-info"
    role="alert"
  >
    <h5
      class="mb-1 font-medium leading-none tracking-tight"
    >
      Alert info
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      You should pay attention to me
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert > has neutral variant 1`] = `
<DocumentFragment>
  <div
    class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg]:absolute [&>svg]:left-4 [&>svg]:text-inherit bg-neutral text-neutral-content border-neutral"
    role="alert"
  >
    <h5
      class="mb-1 font-medium leading-none tracking-tight"
    >
      Alert neutral
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      You should pay attention to me
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert > has no outline 1`] = `
<DocumentFragment>
  <div
    class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg]:absolute [&>svg]:left-4 [&>svg]:text-inherit bg-default text-default-content border-default"
    role="alert"
  >
    <h5
      class="mb-1 font-medium leading-none tracking-tight"
    >
      Alert has no outline
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      You should pay attention to me
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert > has outline 1`] = `
<DocumentFragment>
  <div
    class="relative w-full rounded-lg p-4 [&>svg~*]:pl-7 [&>svg]:absolute [&>svg]:left-4 [&>svg]:text-inherit border-default border bg-default text-default"
    role="alert"
  >
    <h5
      class="mb-1 font-medium leading-none tracking-tight"
    >
      Alert has outline
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      You should pay attention to me
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert > has primary variant 1`] = `
<DocumentFragment>
  <div
    class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg]:absolute [&>svg]:left-4 [&>svg]:text-inherit bg-primary text-primary-content border-primary"
    role="alert"
  >
    <h5
      class="mb-1 font-medium leading-none tracking-tight"
    >
      Alert primary
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      You should pay attention to me
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert > has secondary variant 1`] = `
<DocumentFragment>
  <div
    class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg]:absolute [&>svg]:left-4 [&>svg]:text-inherit bg-secondary text-secondary-content border-secondary"
    role="alert"
  >
    <h5
      class="mb-1 font-medium leading-none tracking-tight"
    >
      Alert secondary
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      You should pay attention to me
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert > has success variant 1`] = `
<DocumentFragment>
  <div
    class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg]:absolute [&>svg]:left-4 [&>svg]:text-inherit bg-success text-success-content border-success"
    role="alert"
  >
    <h5
      class="mb-1 font-medium leading-none tracking-tight"
    >
      Alert success
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      You should pay attention to me
    </div>
  </div>
</DocumentFragment>
`;

exports[`Alert > has warning variant 1`] = `
<DocumentFragment>
  <div
    class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg]:absolute [&>svg]:left-4 [&>svg]:text-inherit bg-warning text-warning-content border-warning"
    role="alert"
  >
    <h5
      class="mb-1 font-medium leading-none tracking-tight"
    >
      Alert warning
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      You should pay attention to me
    </div>
  </div>
</DocumentFragment>
`;
